jsonforms-tooling-common/src/index.ts

```diff
@@ -9,85 +9,110 @@ import { sep } from 'path';
 
 /*
- * Clones a git repository and runs npm install on it
- * @param {string} repo the name of the repo that should be cloned
- * @param {string} path to the folder, where the repo should be cloned into
- * @param {function} callback forwards the current status to the caller
+ * Receives the data from the editor and calls the install methos
+ * @param {any} editorInstance the instance of the editor
+ * @param {any} args the arguments passed to the editor call
+ * @param {string} project the project, that should be installed
  */
-export const cloneAndInstall = (
-  repo: string,
-  path: string,
-  callback: (id: string, result: string, type?: string) => void,
-  name?: string) => {
-  let url = '';
-  switch (repo) {
-    case 'example':
-      url = 'https://github.com/eclipsesource/make-it-happen-react';
-      break;
-    case 'seed':
-      url = 'https://github.com/eclipsesource/jsonforms-react-seed';
-      break;
-    default:
-      return;
+export const createProject = (editorInstance: any, args: any, project: string) => {
+  if (!args) {
+    editorInstance.window.showOpenDialog(editorInstance.OpenDialogOptions = {
+      canSelectMany: false,
+      canSelectFolders: true,
+      canSelectFiles: false,
+      openLabel: 'Select folder',
+    }).then((fileUri: any) => {
+      if (fileUri && fileUri[0].fsPath) {
+        asyncCreateProject(editorInstance, fileUri[0].fsPath, project);
+      } else {
+        showMessage(editorInstance, 'Please select a empty folder', 'err');
+        return;
+      }
+    });
+  } else {
+    asyncCreateProject(editorInstance, args.fsPath, project);
   }
-  const git = simplegit();
-  callback('start-cloning', 'Starting to clone repo');
-  git.clone(url, path)
-    .then(() => {
-      callback('finished-cloning', 'Finished to clone repo');
-      callback('npm-install', 'Running npm install');
-      const result = cp.spawnSync(npm, ['install'], {
-        cwd: path,
-      });
-      callback('signal', result.signal);
-    })
-    .catch((err: any) => { callback('error', err.message, 'err'); });
 };
 
 /**
  * Generates the default UI Schema from a json schema
- * @param {string} path path to the json schema file
- * @param {function} callback forwards the current status to the caller
+ * @param {any} editorInstance the instance of the editor
+ * @param {any} args the arguments passed to the editor call
  */
-export const generateUISchema = (
-  path: string,
-  name: string,
-  callback: (id: string, result: string, type?: string) => void) => {
-  // Read JSON Schema file
-  readFile(path, 'utf8', (readError, data) => {
-    if (readError.message) {
-      callback('error', readError.message, 'err');
-      return;
-    }
-
-    const jsonSchema = JSON.parse(data);
-    validateJSONSchema(jsonSchema, (validateError?: string) => {
-      if (validateError) {
-        callback('error', validateError, 'err');
+export const generateUISchema = (editorInstance: any, args: any) => {
```

Comment:
As with `createProject`, `args` is actually only used for determing a file path.